In [12]:
import pandas as pd

updated_df = pd.read_csv('updated_merged_data.csv')
updated_df

Symbol                    Name                  Sector   Price  \
0      MMM              3M Company             Industrials  222.89   
1      AOS         A.O. Smith Corp             Industrials   60.24   
2      ABT     Abbott Laboratories             Health Care   56.27   
3     ABBV             AbbVie Inc.             Health Care  108.48   
4      ACN           Accenture plc  Information Technology  150.51   
..     ...                     ...                     ...     ...   
374    XEL         Xcel Energy Inc               Utilities   42.44   
375    XYL              Xylem Inc.             Industrials   70.24   
376    YUM         Yum! Brands Inc  Consumer Discretionary   76.30   
377    ZBH  Zimmer Biomet Holdings             Health Care  115.53   
378    ZTS                  Zoetis             Health Care   71.51   

     Price_Earnings  Dividend_Yield  Earnings_Share  52_Week_Low  \
0             24.31        2.332862            7.92       259.77   
1             27.76        1.147959            1.70        68.39   
2             22.51        1.908982            0.26        64.60   
3             19.41        2.499560            3.29       125.86   
4             25.47        1.714470            5.44       162.60   
..              ...             ...             ...          ...   
374           18.14        3.391427            2.21        52.22   
375           30.94        1.170079            1.83        76.81   
376           27.25        1.797080            4.07        86.93   
377           14.32        0.794834            9.01       133.49   
378           32.80        0.682372            1.65        80.13   

     52_Week_High    Market_Cap  ...  Price_Sales  Price_Book  \
0         175.490  138721055226  ...     4.390271       11.34   
1          48.925   10783419933  ...     3.575483        6.35   
2          42.280  102121042306  ...     3.740480        3.19   
3          60.050  181386347059  ...     6.291571       26.14   
4         114.820   98765855553  ...     2.604117       10.62   
..            ...           ...  ...          ...         ...   
374        41.160   21559611927  ...     2.520456        1.94   
375        46.860   12915021000  ...     2.726209        5.31   
376        62.850   27003303098  ...     6.313636      212.08   
377       108.170   24454698119  ...     3.164895        2.39   
378        52.000   35991109776  ...     9.280896       18.09   

                                           SEC Filings Total_ESG_Risk_score  \
0    http://www.sec.gov/cgi-bin/browse-edgar?action...                 37.3   
1    http://www.sec.gov/cgi-bin/browse-edgar?action...                 25.4   
2    http://www.sec.gov/cgi-bin/browse-edgar?action...                 24.8   
3    http://www.sec.gov/cgi-bin/browse-edgar?action...                 29.9   
4    http://www.sec.gov/cgi-bin/browse-edgar?action...                  9.8   
..                                                 ...                  ...   
374  http://www.sec.gov/cgi-bin/browse-edgar?action...                 26.5   
375  http://www.sec.gov/cgi-bin/browse-edgar?action...                 18.1   
376  http://www.sec.gov/cgi-bin/browse-edgar?action...                 20.1   
377  http://www.sec.gov/cgi-bin/browse-edgar?action...                 26.0   
378  http://www.sec.gov/cgi-bin/browse-edgar?action...                 18.8   

     Environment_Risk_Score  Governance_Risk_Score  Social_Risk_Score  \
0                      17.2                    6.5               13.6   
1                       7.2                    6.4               11.9   
2                       2.3                    8.3               14.2   
3                       2.4                   10.4               17.2   
4                       0.8                    4.4                4.6   
..                      ...                    ...                ...   
374                    12.7                    4.5                9.3   
375                     4.3     

## Question 1: Is there a significant correlation between Market Cap and Total ESG Risk Score?

**Null Hypothesis (H0):** There is no significant correlation between Market Cap and Total ESG Risk Score.

**Test Used:** Pearson Correlation

**Results:**
- Correlation coefficient: 0.094
- p-value: 0.067

**Conclusion:**
Since the p-value is greater than 0.05, we fail to reject the null hypothesis. Therefore, we conclude that there is no significant correlation between Market Cap and Total ESG Risk Score.


In [13]:
import scipy.stats as stats

# Calculate the Pearson correlation coefficient and p-value
corr, p_value_corr = stats.pearsonr(updated_df['Market Cap'], updated_df['total_esg_risk_score'])
print(f'Correlation between Market Cap and Total ESG Risk Score: {corr}, p-value: {p_value_corr}')


Correlation between Market Cap and Total ESG Risk Score: 0.09428074376089259, p-value: 0.06673221849881873


## Question 2: Does the Environment Risk Score have a significant impact on Earnings/Share?

**Null Hypothesis (H0):** Environment Risk Score has no significant impact on Earnings/Share.

**Test Used:** Linear Regression

**Results:**
- Coefficient for Environment Risk Score: -0.148
- p-value: 0.003

**Conclusion:**
Since the p-value is less than 0.05, we reject the null hypothesis. Therefore, we conclude that the Environment Risk Score has a significant impact on Earnings/Share.


In [15]:
import statsmodels.formula.api as smf

# Check and replace column names if necessary
updated_df.rename(columns={
    'Earnings/Share': 'Earnings_Share',
    'Environment Risk Score': 'Environment_Risk_Score'
}, inplace=True)

# Perform linear regression analysis
model_1 = smf.ols('Earnings_Share ~ Environment_Risk_Score', data=updated_df).fit()
print(model_1.summary())


                            OLS Regression Results                            
Dep. Variable:         Earnings_Share   R-squared:                       0.023
Model:                            OLS   Adj. R-squared:                  0.020
Method:                 Least Squares   F-statistic:                     8.762
Date:                Fri, 28 Jun 2024   Prob (F-statistic):            0.00327
Time:                        00:16:50   Log-Likelihood:                -1143.4
No. Observations:                 379   AIC:                             2291.
Df Residuals:                     377   BIC:                             2299.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                             coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------------
Intercept                  4

## Question 3: Are the mean Governance Risk Scores different across different sectors?

**Null Hypothesis (H0):** The mean Governance Risk Scores are the same across all sectors.

**Test Used:** ANOVA

**Results:**
- F-statistic: 32.52
- p-value: 6.27e-45

**Conclusion:**
Since the p-value is less than 0.05, we reject the null hypothesis. Therefore, we conclude that the mean Governance Risk Scores are different across different sectors.


In [17]:
import statsmodels.api as sm

# Perform ANOVA test
model_2 = smf.ols('Governance_Risk_Score ~ C(Sector)', data=updated_df).fit()
anova_table = sm.stats.anova_lm(model_2, typ=2)
print(anova_table)


               sum_sq     df          F        PR(>F)
C(Sector)  838.198681   10.0  32.519075  6.271842e-45
Residual   948.542080  368.0        NaN           NaN


In [20]:
# Convert ESG_Risk_Percentile to numeric values
updated_df['ESG_Risk_Percentile'] = updated_df['ESG_Risk_Percentile'].str.extract('(\d+)').astype(float)

# Verify the updated ESG_Risk_Percentile column
print("Updated ESG_Risk_Percentile column:")
print(updated_df['ESG_Risk_Percentile'].head())

Updated ESG_Risk_Percentile column:
0    89.0
1    51.0
2    48.0
3    69.0
4     3.0
Name: ESG_Risk_Percentile, dtype: float64


## Question 4: Is there a significant difference in Dividend Yield between companies with high and low ESG Risk Percentiles?

**Null Hypothesis (H0):** There is no significant difference in Dividend Yield between companies with high and low ESG Risk Percentiles.

**Test Used:** T-test

**Results:**
- t-statistic: 2.39
- p-value: 0.017

**Conclusion:**
Since the p-value is less than 0.05, we reject the null hypothesis. Therefore, we conclude that there is a significant difference in Dividend Yield between companies with high and low ESG Risk Percentiles.


In [22]:
median_esg_percentile = updated_df['ESG_Risk_Percentile'].median()
updated_df['ESG_Class'] = ['High' if x >= median_esg_percentile else 'Low' for x in updated_df['ESG_Risk_Percentile']]
high_esg = updated_df[updated_df['ESG_Class'] == 'High']['Dividend_Yield']
low_esg = updated_df[updated_df['ESG_Class'] == 'Low']['Dividend_Yield']
t_stat, p_value_ttest = stats.ttest_ind(high_esg, low_esg)
print(f'T-test for Dividend Yield between high and low ESG Risk Percentiles: t-statistic: {t_stat}, p-value: {p_value_ttest}')


T-test for Dividend Yield between high and low ESG Risk Percentiles: t-statistic: 2.38775575439674, p-value: 0.01744337401254301


## Question 5: Does the Social Risk Score significantly predict the Price/Earnings ratio?

**Null Hypothesis (H0):** Social Risk Score does not significantly predict the Price/Earnings ratio.

**Test Used:** Linear Regression

**Results:**
- Coefficient for Social Risk Score: -0.156
- p-value: 0.802

**Conclusion:**
Since the p-value is greater than 0.05, we fail to reject the null hypothesis. Therefore, we conclude that the Social Risk Score does not significantly predict the Price/Earnings ratio.


In [23]:
model_3 = smf.ols('Price_Earnings ~ Social_Risk_Score', data=updated_df).fit()
print(model_3.summary())

                            OLS Regression Results                            
Dep. Variable:         Price_Earnings   R-squared:                       0.000
Model:                            OLS   Adj. R-squared:                 -0.002
Method:                 Least Squares   F-statistic:                   0.06308
Date:                Fri, 28 Jun 2024   Prob (F-statistic):              0.802
Time:                        00:22:28   Log-Likelihood:                -1966.8
No. Observations:                 379   AIC:                             3938.
Df Residuals:                     377   BIC:                             3945.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                        coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------
Intercept            27.1361      6.09